### PR TITLE
rename internal components

### PR DESCRIPTION
**Pull request by drahnr**
Ref #132 

Should be merged after #131 is merged, to avoid additional friction.

### CODE REVIEW

**Review by montekki — APPROVED** [review]

**Comment by drahnr** [comment]
Hold on, I am almost done rebasing :)

File: tests/dist.rs [hunk 1]
```diff
@@ -13,8 +13,8 @@ use crate::harness::{
 use async_trait::async_trait;
 use cachepot::config::HTTPUrl;
 use cachepot::dist::{
-    AssignJobResult, CompileCommand, InputsReader, JobId, JobState, RunJobResult, ServerIncoming,
-    ServerOutgoing, SubmitToolchainResult, Toolchain, ToolchainReader,
+    AssignJobResult, CompileCommand, CoordinatorIncoming, CoordinatorOutgoing, InputsReader, JobId,
+    JobState, RunJobResult, SubmitToolchainResult, Toolchain, ToolchainReader,
```
Comment (montekki) [inline tests/dist.rs:17]: `dist::Server` should become `dist::Worker` not `dist::Coordinator`?

File: src/dist/mod.rs [hunk 2]
```diff
@@ -678,22 +677,22 @@ pub trait ServerIncoming: Send + Sync {
     // From Client
     async fn handle_submit_toolchain(
         &self,
-        requester: &dyn ServerOutgoing,
+        requester: &dyn CoordinatorOutgoing,
```
Comment (montekki) [inline src/dist/mod.rs:680]: `WorkerOutgoing`?

File: src/dist/mod.rs [hunk 3]
```diff
@@ -648,27 +647,27 @@ pub trait SchedulerIncoming: Send + Sync {
...
     ) -> ExtResult<UpdateJobStateResult, Error>;
     // From anyone
     fn handle_status(&self) -> ExtResult<SchedulerStatusResult, Error>;
 }
 
-#[cfg(feature = "dist-server")]
+#[cfg(feature = "dist-worker")]
 #[async_trait]
-pub trait ServerIncoming: Send + Sync {
+pub trait CoordinatorIncoming: Send + Sync {
```
Comment (montekki) [inline src/dist/mod.rs:670]: `WorkerIncoming`?

**Review by montekki — COMMENTED** [review]
I think that some renames from `dist::Server` to `dist::Coordinator` were wrong and should instead be to `dist::Worker`.

**Review by montekki — APPROVED** [review]

File: docs/book/src/dist/dist.md [hunk 4]
```diff
@@ -96,17 +96,17 @@ Now generate a token for the server, giving the IP and port the scheduler and cl
...
 ```
 
 *or:*
 
 ```sh
-cachepot-dist auth generate-jwt-hs256-server-token \
+cachepot-dist auth generate-jwt-hs256-coordinator-token \
     --config /path/to/scheduler-config.toml \
-    --server 192.168.1.10:10501
+    --coordinator 192.168.1.10:10501
```
Comment (drahnr) [inline docs/book/src/dist/dist.md:109]: I think this is still incorrect.

File: docs/book/src/dist/dist.md [hunk 5]
```diff
@@ -65,7 +65,7 @@ There are three axes of security in this setup:
 2. Is the client permitted to submit and run jobs?
 3. Can third parties see and/or modify traffic?
 
-### Server Trust
+### Worker Trust
 
 If a server is malicious, they can return malicious compilation output to a user.
 To protect against this, servers must be authenticated to the scheduler. You have three
```
Comment (Xanewok) [inline docs/book/src/dist/dist.md:71]:
```suggestion
To protect against this, build servers must be authenticated to the scheduler. You have three
```

File: docs/book/src/dist/dist.md (hunk 4 above)
Comment (Xanewok) [reply docs/book/src/dist/dist.md:109]: This refers to dist-servers rather than coordinators, right?

File: docs/book/src/dist/dist.md [hunk 6]
```diff
@@ -285,10 +285,10 @@ the heartbeat. If a client does not have the appropriate certificate for communi
 securely with a server (after receiving a job allocation from the scheduler), the
 certificate will be requested from the scheduler.
 
-# Building the Distributed Server Binaries
+# Building the Distributed Worker Binaries
 
 Until these binaries [are included in releases](https://github.com/paritytech/cachepot/issues/393) I've put together a Docker container that can be used to easily build a release binary:
 
 ```toml
```
Comment (Xanewok) [inline docs/book/src/dist/dist.md:292]:
```suggestion
```sh (or none?)
```

File: docs/book/src/dist/quickstart.md [hunk 7]
```diff
@@ -17,7 +17,7 @@ If you're only planning to use the client, it is enabled by default, so just `ca
 
 ## Configure a scheduler
 
-If you're adding a server to a cluster that has already be set up, skip ahead to [configuring a build server](#configure-a-build-server).
+If you're adding a server to a cluster that has already be set up, skip ahead to [configuring a build server](#configure-a-build-coordinator).
```
Comment (Xanewok) [inline docs/book/src/dist/quickstart.md:20]: This should refer to build servers aka dist workers

File: src/test/tests.rs [hunk 8]
```diff
@@ -35,7 +35,7 @@ use tokio::runtime::Runtime;
 
 /// Options for running the server in tests.
 #[derive(Default)]
-struct ServerOptions {
+struct WorkerOptions {
```
Comment (Xanewok) [inline src/test/tests.rs:38]:
```suggestion
struct CoordinatorOptions {
```

File: src/test/tests.rs [hunk 9]
```diff
@@ -46,7 +46,7 @@ struct ServerOptions {
 ///
 /// * The port on which the server is listening.
```
Comment (Xanewok) [inline src/test/tests.rs:47]:
```suggestion
/// * The port on which the coordinator is listening.
```

File: src/test/tests.rs [hunk 10]
```diff
@@ -55,12 +55,12 @@ fn run_server_thread<T>(
...
 ) -> (
     u16,
-    Sender<ServerMessage>,
+    Sender<CoordinatorMessage>,
     Arc<Mutex<MockCommandCreator>>,
     thread::JoinHandle<()>,
 )
 where
-    T: Into<Option<ServerOptions>> + Send + 'static,
+    T: Into<Option<WorkerOptions>> + Send + 'static,
```
Comment (Xanewok) [inline src/test/tests.rs:63]:
```suggestion
    T: Into<Option<CoordinatorOptions>> + Send + 'static,
```

File: src/test/tests.rs [hunk 11]
```diff
@@ -115,13 +115,13 @@ fn test_server_shutdown_no_idle() {
     // Set a ridiculously low idle timeout.
     let (port, _sender, _storage, child) = run_server_thread(
         f.tempdir.path(),
-        ServerOptions {
+        WorkerOptions {
```
Comment (Xanewok) [inline src/test/tests.rs:118]:
```suggestion
        CoordinatorOptions {
```

File: src/test/tests.rs [hunk 12]
```diff
@@ -132,7 +132,7 @@ fn test_server_idle_timeout() {
     // Set a ridiculously low idle timeout.
     let (_port, _sender, _storage, child) = run_server_thread(
         f.tempdir.path(),
-        ServerOptions {
+        WorkerOptions {
```
Comment (Xanewok) [inline src/test/tests.rs:135]:
```suggestion
        CoordinatorOptions {
```

**Review by Xanewok — CHANGES_REQUESTED** [review]
Marked some yet unchaged mentions of server

**Comment by Xanewok** [comment]
The more I think about it, the more I realize how overloaded the term "server" is in cachepot/sccache context :( This rename helps a lot but I'm still wondering if "coordinator" is a good choice here. Maybe we could use "daemon" here? It's more recognizable in terms of build systems (e.g. Gradle, Buck, Bazel) and I think the benefits from familiarity are bigger, even if our "daemon" can sometimes be ran on the foreground, if that makes sense.

File: docs/book/src/dist/dist.md (hunk 4 above)
Comment (drahnr) [reply docs/book/src/dist/dist.md:109]: Yep

File: docs/book/src/dist/dist.md (hunk 6 above)
Comment (drahnr) [reply docs/book/src/dist/dist.md:292]: Yep

File: docs/book/src/dist/dist.md (hunk 5 above)
Comment (drahnr) [reply docs/book/src/dist/dist.md]: Workers

**Comment by montekki** [comment]
The CI looks ok, I think that all review points have been addressed so merging now

**Pull request merged**